recipe_scrapers/plugins/schemaorg_fill.py

```diff
@@ -57,7 +57,9 @@ def decorated_method_wrapper(self, *args, **kwargs):
                 return decorated(self, *args, **kwargs)
             except (FillPluginException, NotImplementedError) as e:
                 function = getattr(self.schema, decorated.__name__)
-                if self.schema.data and function:
+                if not self.schema.data:
+                    raise RecipeSchemaNotFound(url=self.url)
```

Comment:
Question: should we raise an exception here (quite a large behaviour change), or instead perhaps simply return `None`?

Reply:
Returning `None` has the least likelihood of breaking any downstream uses however I don't think this is a reason not to go with this implementation. I think the swap to offering a custom exception makes the output much clearer and could limit some frustration with future debugging.

Personally I am in favor of this PR's implementation.

Reply:
Thanks @jknndy - ok, yep, that seems good to me too.  After reflecting on it, returning `None` might provide a false sense to downstream consumers that scraping is working, when in fact there is some easily-discoverable problem about the input data.

One other question is whether we could/should somehow bring both `RecipeSchemaNotFound` and the existing `NoSchemaFoundInWildMode` exceptions into alignment (same inheritance tree, for example).  They seem fairly similar, but at the moment `NoSchemaFoundInWildMode` does not inherit from `SchemaOrgException`.  Changing it so that it does also inherit from that would (also) be an interface-breaking change, I think.

I'll likely merge this later today.

Reply:
> I'll likely merge this later today.

(but to clarify: I don't think I'll apply the potential `NoSchemaFoundInWildMode` base-class change as part of that merge.  it's just an idea at this point)